File: examples/PHP/tasksink.php
<?php
/*
 *  Task sink
 *  Binds PULL socket to tcp://localhost:5558
 *  Collects results from workers via that socket
 * @author Ian Barber <ian(dot)barber(at)gmail(dot)com>
 */

//  Prepare our context and socket
$context = new ZMQContext();
$receiver = new ZMQSocket($context, ZMQ::SOCKET_PULL);
$receiver->bind("tcp://*:5558");

//  Wait for start of batch
$string = $receiver->recv();

//  Start our clock now
$tstart = microtime(true);

//  Process 100 confirmations
$total_msec = 0;     //  Total calculated cost in msecs
for ($task_nbr = 0; $task_nbr < 100; $task_nbr++) {
	$string = $receiver->recv();
	if($task_nbr % 10 == 0) {
		echo ":";
	} else {
		echo ".";
	}
}

$tend = microtime(true);

$total_msec = ($tend - $tstart) * 1000;
echo PHP_EOL;
printf ("Total elapsed time: %d msec", $total_msec);
echo PHP_EOL;

File: examples/Rust/tasksink.rs
#![crate_name = "tasksink"]

extern crate zmq;

use std::io::{self, Write};
use std::time::Instant;

fn main() {
    let context = zmq::Context::new();
    let receiver = context.socket(zmq::PULL).unwrap();
    assert!(receiver.bind("tcp://*:5558").is_ok());

    let _ = receiver.recv_string(0).unwrap();

    let start_time = Instant::now();

    for task_nbr in 0..100 {
        let _ = receiver.recv_string(0).unwrap();
        if task_nbr % 10 == 0 {
            println!(":");
        } else {
            println!(".");
        }
        let _ = io::stdout().flush();
    }

    println!("Total elapsed time: {:?} msec", start_time.elapsed());

    let control = context.socket(zmq::PUB).unwrap();
    assert!(control.bind("tcp://*:5559").is_ok());
    control.send_str("kill", 0).unwrap();
}
